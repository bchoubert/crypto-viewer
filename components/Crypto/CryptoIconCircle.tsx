import Colors from "@/assets/Colors";
import { FC, ReactNode, memo, useMemo } from "react";
import { StyleSheet, View } from "react-native";

interface CryptoIconCircleProps {
  children: ReactNode;
  color: string;
}

const CryptoIconCircle: FC<CryptoIconCircleProps> = memo(({
  children,
  color,
}) => {
  const styles = useMemo(() => StyleSheet.create({
    container: {
      backgroundColor: color,
      color: Colors.white,
      height: 36,
      width: 36,
      padding: 5,
      borderRadius: 18,
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center'
    }
  }), [color]);

  return (
    <View style={styles.container}>
      {children}
    </View>
  );
});

export default CryptoIconCircle;
