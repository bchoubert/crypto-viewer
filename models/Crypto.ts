// Currency type as returned by the API
export default interface Crypto {
  id: string,
  name: string,
  status: 'online' | 'offline',
  details: {
    type: string,
    symbol: string
  },
  price?: number
}
