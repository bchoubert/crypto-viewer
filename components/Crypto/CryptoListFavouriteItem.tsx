import { SettingsContext } from "@/contexts/settings.provider";
import { ICrypto, quoteDetails } from "@/types/crypto.types";
import { FC, memo, useContext, useMemo } from "react";
import CryptoIcon from "../CryptoIcon";
import ECrypto from "@/constants/cryptos.enum";
import CryptoDetails from "@/constants/cryptodetails.constants";
import Colors, { EColor } from "@/assets/Colors";
import { Link } from "expo-router";
import { Pressable, StyleSheet, Text, View } from "react-native";
import CryptoIconCircle from "./CryptoIconCircle";
import { printNumber } from "@/services/print.service";

interface CryptoListFavouriteItemProps {
  item: ICrypto;
}

const styles = StyleSheet.create({
  container: {
    height: 60,
    width: '46%',
    marginLeft: '2%',
    marginRight: '2%',
    marginTop: 10,
    marginBottom: 10,
    borderRadius: 10,
    borderBottomColor: Colors.lightGray,
    borderBottomWidth: 1,
    paddingLeft: 10,
    paddingRight: 10,
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: Colors.white,
  },
  iconplaceholder: {
    width: 36,
  },
  left: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8,
  },
  right: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    gap: 5
  },
  icon: {
    height: 20,
    width: 20,
  },
  titles: {
    display: 'flex',
    flexDirection: 'column',
    color: Colors.white,
  },
  title: {
    color: Colors.white,
    fontWeight: '500',
  },
  subtitle: {
    color: Colors.midGray,
  },
  price: {
    color: Colors.white,
  }
});

const CryptoListFavouriteItem: FC<CryptoListFavouriteItemProps> = memo(({
  item
}) => {
  const { settings } = useContext(SettingsContext);

  const CryptoIconInstance = useMemo(() => CryptoIcon[item.id.toLowerCase() as ECrypto], [item.id]);
  const cryptoDetails = useMemo(() => CryptoDetails[item.id.toLowerCase() as ECrypto], [item.id]);

  const [color, quoteSymbol, statusColor] = useMemo(() => {
    const baseColor = cryptoDetails?.color || Colors.darkGray;
    const quoteSymbol = quoteDetails[settings.quote]?.symbol;
    const statusColor = item.status === 'online' ? EColor.green : EColor.red;

    return [
      baseColor,
      quoteSymbol,
      statusColor,
    ]
  }, [cryptoDetails, item]);

  return (
    <Link href={`/crypto/${item.id}`} style={styles.container} asChild>
      <Pressable style={{ ...styles.container, backgroundColor: color }}>
        <View style={styles.left}>
          {CryptoIconInstance ? (
            <CryptoIconCircle color={Colors.white} statusColor={statusColor}>
              <CryptoIconInstance style={styles.icon} fill={color} />
            </CryptoIconCircle>
          ) : (<View style={styles.iconplaceholder} />)}
          <View style={styles.titles}>
            <Text style={styles.title}>{item.name}</Text>  
            <Text style={styles.subtitle}>{item.id}</Text> 
          </View>
        </View>
        <View style={styles.right}>
          {item.rate ? (
            <Text style={styles.price}>{printNumber(item.rate, quoteSymbol)}</Text>
          ) : null}
        </View>
      </Pressable>
    </Link>
  );
});

export default CryptoListFavouriteItem;