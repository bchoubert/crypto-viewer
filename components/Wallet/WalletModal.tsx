import React, { FC, memo, useMemo } from 'react';
import { Picker } from '@react-native-community/picker';
import { Modal, TextInput, TouchableHighlight, Text, View, StyleSheet, Dimensions } from 'react-native';

import Colors from '../../assets/Colors';

import UtilsService from '../../services/Utils.service';

import Crypto from '../../models/Crypto';
import CryptoIcon from '../Utils/CryptoIcon';

const styles = StyleSheet.create({
  modal_container: {
    flex: 1,
    backgroundColor: 'rgba(0, 0, 0, .7)',
    justifyContent: 'flex-end',
    alignItems: 'center'
  },
  modal: {
    width: Dimensions.get('window').width,
    height: 220,
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
    backgroundColor: '#FFFFFF'
  },
  topBar: {
    height: 60,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: Colors.lightGray,
    borderTopLeftRadius: 15,
    borderTopRightRadius: 15,
    paddingLeft: 15,
    paddingRight: 15
  },
  modal_title: {
    fontSize: 18
  },
  modal_action: {
    color: Colors.blue,
    fontSize: 18,
    minWidth: 50,
    textAlign: 'left'
  },
  modal_main_action: {
    fontWeight: 'bold',
    textAlign: 'right'
  },
  modal_label: {
    fontSize: 10,
    color: Colors.gray
  },
  modal_content: {
    padding: 10
  },
  disabled_modal_action: {
    color: Colors.gray,
    fontWeight: 'normal',
  },
  modal_crypto: {
    flex: 1
  },
  amount_input: {
    padding: 5,
    borderBottomColor: Colors.gray,
    borderBottomWidth: 1
  },
  modal_crypto_container: {
    flexDirection: 'row',
    alignItems: 'center',
    height: 50,
    flexBasis: 50
  },

});

interface WalletModalProps {
  isOpen: boolean;
  closeModal: () => any;
  selectedAmount: number;
  selectedCryptoKey: string;
  saveWallet: () => any;
  onSelectedCryptoKeyChange: (newCryptoKey: string) => any;
  onSelectedAmountChange: (newAmount: string) => any;
  cryptos: Crypto[];
}

const WalletModal: FC<WalletModalProps> = ({
  isOpen,
  closeModal,
  selectedAmount,
  selectedCryptoKey,
  saveWallet,
  onSelectedCryptoKeyChange,
  onSelectedAmountChange,
  cryptos,
}) => {

  // Validate function for the modal form
  const isValid = useMemo(
    () => !!selectedAmount && !!selectedCryptoKey,
    [selectedAmount, selectedCryptoKey],
  );

  const cryptoOptions = useMemo(
    () => cryptos.map(asset => <Picker.Item key={asset.id} label={asset.name} value={asset.id} />),
    [cryptos],
  );

  return (
    <Modal
      animationType="slide"
      visible={isOpen}
      transparent={true}
    >
      <View style={styles.modal_container}>
        <View style={styles.modal}>
          <View style={styles.topBar}>
            {/* Cancel button */}
            <TouchableHighlight onPress={closeModal}>
              <Text style={styles.modal_action}>Cancel</Text>
            </TouchableHighlight>
            <Text style={styles.modal_title}>Add to your Wallet</Text>
            {/* Save buttom, active only if isValid() returnes true */}
            <TouchableHighlight onPress={saveWallet}>
              <Text style={{ ...styles.modal_action, ...styles.modal_main_action, ...(isValid && [] || styles.disabled_modal_action) }}>Save</Text>
            </TouchableHighlight>
          </View>
          <View style={styles.modal_content}>
            <Text style={styles.modal_label}>Crypto</Text>
            <View style={styles.modal_crypto_container}>
              {!!selectedCryptoKey &&
                <CryptoIcon code={selectedCryptoKey.toLowerCase()} />
              }
              {/* Picker with a default item */}
              <Picker
                selectedValue={selectedCryptoKey}
                style={styles.modal_crypto}
                onValueChange={onSelectedCryptoKeyChange}>
                <Picker.Item label='Please select an option...' value={null} />
                {cryptoOptions}
              </Picker>
            </View>
            <Text style={styles.modal_label}>Amount</Text>
            <TextInput
              placeholder="1.02"
              underlineColorAndroid='transparent'
              keyboardType={'numeric'}
              style={styles.amount_input}
              value={!!selectedAmount ? (selectedAmount + '') : null}
              onChangeText={onSelectedAmountChange} />
          </View>
        </View>
      </View>
    </Modal>
  );
};

export default memo(WalletModal);
