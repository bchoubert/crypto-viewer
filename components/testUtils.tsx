/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderAPI } from '@testing-library/react-native';
import '@testing-library/jest-dom/extend-expect';
import React, { ReactElement } from 'react';

import NavigationProvider from '../contexts/NavigationProvider';
import { SettingsContext } from '../contexts/SettingsProvider';
import ThemeProvider from '../contexts/ThemeProvider';
import TranslationProvider from '../contexts/TranslationProvider';

import { defaultSettings } from '../models/SettingsType';

jest.useFakeTimers();

export const enrichProviders = (node: ReactElement): ReactElement => (
  <SettingsContext.Provider
    value={{
      settings: defaultSettings,
      changeSettings: jest.fn(),
    }}
  >
    <TranslationProvider>
      <ThemeProvider>
        <NavigationProvider>
          {node}
        </NavigationProvider>
      </ThemeProvider>
    </TranslationProvider>
  </SettingsContext.Provider>
);

// eslint-disable-next-line max-len
export const renderNode = (node: ReactElement): RenderAPI => render(enrichProviders(node));
