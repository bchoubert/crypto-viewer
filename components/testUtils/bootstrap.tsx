/* eslint-disable no-console */
/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderAPI } from '@testing-library/react-native';
import '@testing-library/jest-dom/extend-expect';
import React, { ReactElement } from 'react';
import * as PassThoughComponents from '../PassThroughComponents';

import NavigationProvider from '../../contexts/NavigationProvider';
import { SettingsContext } from '../../contexts/SettingsProvider';
import ThemeProvider from '../../contexts/ThemeProvider';
import TranslationProvider from '../../contexts/TranslationProvider';

import { defaultSettings } from '../../models/SettingsType';
import consoleMocker from './consoleMocker';

jest.useFakeTimers();

export const enrichProviders = (node: ReactElement): ReactElement => (
  <SettingsContext.Provider
    value={{
      settings: defaultSettings,
      changeSettings: jest.fn(),
    }}
  >
    <TranslationProvider>
      <ThemeProvider>
        <NavigationProvider>
          {node}
        </NavigationProvider>
      </ThemeProvider>
    </TranslationProvider>
  </SettingsContext.Provider>
);

// eslint-disable-next-line max-len
export const renderNode = (node: ReactElement): RenderAPI => render(enrichProviders(node));

beforeAll(() => {
  consoleMocker.mockConsole();
  // eslint-disable-next-line no-extend-native
  Date.prototype.getTimezoneOffset = jest.fn(() => 0);
});

afterAll(() => {
  consoleMocker.restoreConsole();
  jest.resetAllMocks();
});

jest.spyOn(PassThoughComponents, 'Image');
(PassThoughComponents.Image as unknown as jest.Mock).mockImplementation(() => (<div>IMAGE</div>));
