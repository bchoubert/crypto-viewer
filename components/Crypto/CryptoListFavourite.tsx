import { ICryptoFavourite } from "@/types/crypto.types";
import { FC, memo } from "react";
import { StyleSheet, View } from "react-native";
import CryptoListFavouriteItem from "./CryptoListFavouriteItem";
import Colors from "@/assets/Colors";

interface CryptoListFavouriteProps {
  item: ICryptoFavourite;
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: Colors.white,
  }
});

const CryptoListFavourite: FC<CryptoListFavouriteProps> = memo(({
  item
}) => (
  <View style={styles.container}>
    {item.cryptos.map(i => (
      <CryptoListFavouriteItem key={i.id} item={i} />
    ))}
  </View>
));

export default CryptoListFavourite;