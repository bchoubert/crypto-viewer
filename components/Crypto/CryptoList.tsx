import NetworkService from "@/services/network.service";
import { FC, memo, useCallback, useContext, useEffect, useMemo, useState } from "react";
import { SectionList, SectionListData, StyleSheet, ToastAndroid } from "react-native";
import CryptoListItem from "./CryptoListItem";
import { ICrypto, ICryptoFavourite } from "@/types/crypto.types";
import CryptoListSection from "./CryptoListSection";
import { SettingsContext } from "@/contexts/settings.provider";
import { TranslationsContext } from "@/contexts/translations.provider";
import { showError } from "@/services/error.service";
import CryptoListFavourite from "./CryptoListFavourite";

const CryptoList: FC = memo(() => {
  const { settings, hasFavourite } = useContext(SettingsContext);
  const translation = useContext(TranslationsContext);

  const [isLoading, setLoading] = useState(true);

  const [cryptosWithRates, setCryptosWithRate] = useState<ICrypto[]>([]);
  const [cryptosWithoutRates, setCryptosWithoutRate] = useState<ICrypto[]>([]);

  const fetchCryptos = useCallback(async () => {
    setLoading(true);

    const [fetchedCurrencies, exchangeRatesResponse] = await Promise.all([
      NetworkService.fetchCryptos().catch(showError),
      NetworkService.fetchCryptoExchangeRates(settings.quote).catch(showError),
    ]);

    if (!fetchedCurrencies) {
      ToastAndroid.show(translation.list.error, ToastAndroid.BOTTOM);
      return [[], []];
    }
    
    const rates = (exchangeRatesResponse?.data?.rates || {});
    const fetchedCurrenciesWithRates: ICrypto[] = fetchedCurrencies.filter(c => c?.details?.type === 'crypto').map(f => ({
      ...f,
      rate: rates[f.id] ? parseFloat(rates[f.id]) : undefined,
    }));
    
    setCryptosWithRate(fetchedCurrenciesWithRates.filter(r => !!r.rate));
    setCryptosWithoutRate(fetchedCurrenciesWithRates.filter(r => !r.rate));
  
    setLoading(false);
  }, [translation]);

  const sections = useMemo(() => {
    if (!cryptosWithRates) {
      return [];
    }

    const favouriteCryptos = cryptosWithRates.filter(c => hasFavourite(c.id));
    const favouriteItems: ICryptoFavourite[] = [];
    favouriteCryptos.forEach((c: ICrypto, i: number) => {
      if (i % 2 === 0) {
        favouriteItems.push({ cryptos: [favouriteCryptos[i], favouriteCryptos[i + 1]].filter(Boolean) })
      }
    });

    console.log(favouriteItems.length)

    return [
      {
        id: 'favourites',
        title: translation.list.favourites,
        data: favouriteItems,
      },
      {
        id: 'main',
        title: translation.list.main,
        data: cryptosWithRates,
      },
      {
        id: 'others',
        title: translation.list.other,
        data: cryptosWithoutRates,
      },
    ];
  }, [cryptosWithRates, cryptosWithoutRates, translation]);

  const handleRenderItem = useCallback(
    (props: { item: ICrypto | ICryptoFavourite, section: { id: string } }) => {

    if (props.section.id === 'favourites') {
      return (<CryptoListFavourite item={props.item as ICryptoFavourite} />);
    }
    
    return (<CryptoListItem item={props.item as ICrypto} />);
  }, []);

  const handleRenderSectionHeader = useCallback(
    (props: { section: { title: string } }) => <CryptoListSection title={props.section.title} />,
    [],
  );

  useEffect(() => {
    fetchCryptos();
  }, []);
  
  return (
    <SectionList
      onRefresh={fetchCryptos}
      refreshing={isLoading}

      sections={sections}
      renderItem={handleRenderItem}
      renderSectionHeader={handleRenderSectionHeader}
      stickySectionHeadersEnabled
    />
  );
});

export default CryptoList;