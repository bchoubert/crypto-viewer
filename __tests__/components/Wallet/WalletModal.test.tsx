import { fireEvent } from '@testing-library/react-native';
import React from 'react';

import { renderNode } from '../../../components/testUtils/bootstrap';
import WalletModal from '../../../components/Wallet/WalletModal';
import Crypto from '../../../models/Crypto';

describe('<WalletModal />', () => {
  const closeModalFn = jest.fn();
  const saveWalletFn = jest.fn();
  const onSelectedCryptoKeyChangeFn = jest.fn();
  const onSelectedAmountChangeFn = jest.fn();

  const commonProps = {
    isOpen: true,
    closeModal: closeModalFn,
    selectedAmount: '3',
    selectedCryptoKey: 'btc',
    isFormValid: true,
    saveWallet: saveWalletFn,
    onSelectedCryptoKeyChange: onSelectedCryptoKeyChangeFn,
    onSelectedAmountChange: onSelectedAmountChangeFn,
    cryptos: [
      { id: 'btc', name: 'Bitcoin' } as Crypto,
      { id: 'ltc', name: 'Litecoin' } as Crypto,
    ],
  };

  it('Prints correctly', () => {
    const { toJSON } = renderNode(
      <WalletModal
        {...commonProps}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('Basic behaviour', async () => {
    const node = renderNode(
      <WalletModal
        {...commonProps}
      />,
    );

    expect(node.getByLabelText('amount-input').props.value).toEqual('3');
    expect(node.getByTestId('crypto-selector').props.selectedIndex).toEqual(1);

    expect(closeModalFn).toHaveBeenCalledTimes(0);
    fireEvent.press(node.getByText('Cancel'));
    expect(closeModalFn).toHaveBeenCalledTimes(1);

    expect(saveWalletFn).toHaveBeenCalledTimes(0);
    fireEvent.press(node.getByText('Save'));
    expect(saveWalletFn).toHaveBeenCalledTimes(1);
  });

  it('Form fulfilment', () => {
    const node = renderNode(
      <WalletModal
        {...commonProps}
      />,
    );

    expect(onSelectedAmountChangeFn).toHaveBeenCalledTimes(0);
    fireEvent.changeText(
      node.getByLabelText('amount-input'),
      '123',
    );
    expect(onSelectedAmountChangeFn).toHaveBeenCalledTimes(1);
    expect(onSelectedAmountChangeFn).toHaveBeenCalledWith('123');

    expect(node.getByLabelText('amount-input').props.value).toEqual('3');
  });
});
