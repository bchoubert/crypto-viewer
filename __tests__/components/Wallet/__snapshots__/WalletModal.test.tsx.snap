// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WalletModal /> Prints correctly 1`] = `
<Modal
  animationType="slide"
  hardwareAccelerated={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, .7)",
        "flex": 1,
        "justifyContent": "flex-end",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderTopLeftRadius": 15,
          "borderTopRightRadius": 15,
          "height": 220,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#F8F8F8",
            "borderTopLeftRadius": 15,
            "borderTopRightRadius": 15,
            "flexDirection": "row",
            "height": 60,
            "justifyContent": "space-between",
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text
            style={
              Object {
                "color": "#005BE3",
                "fontSize": 18,
                "minWidth": 50,
                "textAlign": "left",
              }
            }
          >
            Cancel
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 18,
            }
          }
        >
          Add to your Wallet
        </Text>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <Text
            style={
              Object {
                "color": "#005BE3",
                "fontSize": 18,
                "fontWeight": "bold",
                "minWidth": 50,
                "textAlign": "right",
              }
            }
          >
            Save
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "padding": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 10,
            }
          }
        >
          Crypto
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexBasis": 50,
              "flexDirection": "row",
              "height": 50,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F7931B",
                "borderRadius": 10,
                "display": "flex",
                "justifyContent": "center",
                "marginRight": 10,
                "padding": 7,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "height": 26,
                  "maxHeight": 26,
                  "width": 26,
                }
              }
            >
              btc
            </Text>
          </View>
          <View
            style={
              Object {
                "color": "#222222",
                "flex": 1,
              }
            }
          >
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "Please select an option...",
                    "textColor": undefined,
                    "value": null,
                  },
                  Object {
                    "label": "Bitcoin",
                    "textColor": undefined,
                    "value": "btc",
                  },
                  Object {
                    "label": "Litecoin",
                    "textColor": undefined,
                    "value": "ltc",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={1}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
              testID="crypto-selector"
            />
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#222222",
              "fontSize": 10,
            }
          }
        >
          Amount
        </Text>
        <TextInput
          accessibilityLabel="amount-input"
          allowFontScaling={true}
          keyboardType="numeric"
          onChangeText={[MockFunction]}
          placeholder="1.02"
          rejectResponderTermination={true}
          style={
            Object {
              "borderBottomColor": "#222222",
              "borderBottomWidth": 1,
              "color": "#222222",
              "padding": 5,
            }
          }
          underlineColorAndroid="transparent"
          value="3"
        />
      </View>
    </View>
  </View>
</Modal>
`;
